engine/crates/parser-openapi/src/graph/input_value.rs

```diff
@@ -39,9 +39,12 @@ impl InputValue {
                     unreachable!()
                 };
 
-                // The HasType edge can introduce more wrapping so we need to make sure to account
+                // The HasType edge can introduce list wrapping so we need to make sure to account
                 // for that.
-                let wrapping = wrapping.wrap_with(edge_wrapping.clone());
+                let wrapping = match edge_wrapping {
+                    WrappingType::List(_) => wrapping.wrap_required().wrap_with(edge_wrapping.clone()),
+                    WrappingType::NonNull(_) | WrappingType::Named => wrapping,
```

Comment:
I think there is a blind spot in this logic: if the referenced type is explicitly declared as `nullable: true` or as a union of the type and null, we should mutate `wrapping` to remove the outermost `NonNull` if it exists. Can we assume that `WrappingType::Named` in this match would always correspond to the referenced type being declared as nullable?

Reply:
> we should mutate wrapping to remove the outermost NonNull if it exists

to elaborate on this case with an example: the field is part of `required` on the parent, but the referenced type is nullable. So omitting the field would not be allowed, but passing the field with a `null` value would be. This is hard to model properly in GraphQL...

Reply:
> Can we assume that WrappingType::Named in this match would always correspond to the referenced type being declared as nullable?

I think that's a safe assumption for the HasType edge at least.  Others I am less sure of.

Reply:
Let me check what kind of diff that produces in snapshots and if that makes sense given the api definitions.

Reply:
This:

```
                    WrappingType::Named => wrapping.set_required(false),
                    WrappingType::NonNull(_) => wrapping,
```

produces no difference in snapshots. I guess we don't test references to schema items that are explicitly nullable.